utils/fileops.go

```diff
@@ -0,0 +1,95 @@
+package utils
+
+import (
+	"compress/gzip"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func StoreHTTPResponseToFile(r *http.Response, fileName string, dataFeedName string) error {
+	contentReader := r.Body
+	if IsGzipResponse(r) {
+		var err error
+		contentReader, err = gzip.NewReader(r.Body)
+		if err != nil {
+			log.WithError(err).WithFields(log.Fields{"StatusCode": r.StatusCode, "DataFeedName": dataFeedName}).Error("gzip reader could not read NVD data feed")
+			return err
+		}
+		defer contentReader.Close()
+	}
+
+	f, err := os.OpenFile(fileName, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0666)
+	if err != nil {
+		log.WithError(err).WithField("Filename", fileName).Warning("could not store NVD data feed to filesystem")
+		return err
+	}
+	defer closeFile(f)
+
+	_, err = io.Copy(f, contentReader)
+	if err != nil {
+		log.WithError(err).WithField("Filename", fileName).Warning("could not stream NVD data feed to filesystem")
+		return err
+	}
+	return nil
+}
+
+func StoreCVEsToFile(cves []NvdCVEEntry, fileName, hashFileName string) error {
+	f, err := os.OpenFile(fileName, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0666)
+	if err != nil {
+		log.WithError(err).WithField("Filename", fileName).Warning("could not open file in truncate mode")
+		return err
+	}
+	defer closeFile(f)
+
+	m, err := json.Marshal(cves)
+	if err != nil {
+		log.WithError(err).WithField("Filename", fileName).Errorf("could not marshall %d CVEs", len(cves))
+		return err
+	}
+
+	if _, err = f.Write(m); err != nil {
+		log.WithError(err).WithField("Filename", fileName).Warning("could not write marshalled CVEs list on file", fileName)
+		return err
+	}
+	log.Infof("done writing file: %s", fileName)
+
+	sha256Bytes := sha256.Sum256(m)
+	if err := WriteToFile(hashFileName, hex.EncodeToString(sha256Bytes[:])); err != nil {
+		log.WithError(err).WithField("Filename", hashFileName).Warning("could not write hash on file", hashFileName)
+		return err
+	}
+	log.Infof("done writing file: %s", hashFileName)
+
+	return nil
+}
+
+func WriteToFile(fileName string, data string) error {
+	f, err := os.OpenFile(fileName, os.O_CREATE|os.O_WRONLY, 0666)
```

Comment:
I don't think you need to open the file here. (In fact, I think you can get rid of this function and just call `ioutil.WriteFile` directly at the caller site)